VideoLocker/src/test/java/org/edx/mobile/test/http/ApiTests.java

```diff
@@ -204,4 +228,136 @@ public void testEnrollInACourse() throws Exception {
+            }
+        };
+        HttpManager.HttpResult result = requestDelegate.invokeHttpCall();
+        assertNotNull(result);
+        assertEquals(HttpStatus.SC_OK, result.statusCode);
+        assertNotNull(result.body);
+
+        final CourseComponent courseComponent = requestDelegate.fromJson(result.body);
+        assertNotNull(courseComponent);
+        assertNotNull(courseComponent.getRoot());
+        assertEquals(courseId, courseComponent.getCourseId());
+
+        List<IBlock> children = courseComponent.getChildren();
+        assertNotNull(children);
+        List<CourseComponent> childContainers = new ArrayList<>();
+        List<CourseComponent> childLeafs = new ArrayList<>();
+        for (IBlock c : children) {
+            assertTrue(c instanceof CourseComponent);
+            final CourseComponent child = (CourseComponent) c;
+            assertEquals(child, courseComponent.find(new Filter<CourseComponent>() {
+                @Override
+                public boolean apply(CourseComponent component) {
+                    return child.getId().equals(component.getId());
+                }
+            }));
+            List<IBlock> grandchildren = child.getChildren();
+            for (IBlock gc : grandchildren) {
+                assertTrue(gc instanceof CourseComponent);
+                final CourseComponent grandchild = (CourseComponent) c;
+                assertEquals(grandchild, courseComponent.find(new Filter<CourseComponent>() {
+                    @Override
+                    public boolean apply(CourseComponent component) {
+                        return grandchild.getId().equals(component.getId());
+                    }
+                }));
+            }
+            assertNull(child.find(new Filter<CourseComponent>() {
+                @Override
+                public boolean apply(CourseComponent component) {
+                    return courseComponent.getId().equals(component.getId());
+                }
+            }));
+            if (child.isContainer()) {
+                childContainers.add(child);
+            } else {
+                childLeafs.add(child);
+            }
+        }
+        assertEquals(childContainers, courseComponent.getChildContainers());
+        assertEquals(childLeafs, courseComponent.getChildLeafs());
+
+        assertTrue(courseComponent.isLastChild());
+        int childrenSize = children.size();
+        assertTrue(childrenSize > 0);
+        assertTrue(((CourseComponent)
+                children.get(childrenSize - 1)).isLastChild());
+
+        BlockType blockType = courseComponent.getType();
+        assertSame(courseComponent,
+                courseComponent.getAncestor(Integer.MAX_VALUE));
+        assertSame(courseComponent,
+                courseComponent.getAncestor(EnumSet.of(blockType)));
+
+        List<HasDownloadEntry> videos = courseComponent.getVideos();
+        assertNotNull(videos);
+        for (HasDownloadEntry video : videos) {
```

Comment:
@1zaman  i created a ticket for this https://openedx.atlassian.net/browse/MA-846